### DISCUSSION TITLE
Cache safety with use of cookie

### DISCUSSION THREAD

**Discussion by dnrce** [grafana/grafana-json-datasource#179] [Q&A]
Thanks for a great plugin!

I have a question about use of the cache. I am trying to query an API that requires passthrough of a user's auth cookie to return user-specific data. (The user is anonymous to Grafana.) If I live-update the cookie value, it doesn't seem to invalidate the cache. In this use case, am I at risk of cached results being leaked between users?

**Comment by marcusolsson** [comment]
Firstly, the data is cached in the browser, so users would need to use the same machine. Secondly, the cache is stored in-memory in a [JavaScript object](https://github.com/ptarjan/node-cache/blob/27a1eeec18660b45d39cc5daf44d9647e696e48b/index.js#L4) rather than, say, local storage. If you still have doubts, please contact me at marcus.olsson (at) grafana.com.
